### DISCUSSION TITLE
UGS sender not working on Raspberry Pi 4 running latest 64bit Raspbian OS

### DISCUSSION THREAD

**Discussion by HubertJH** [winder/Universal-G-Code-Sender#2516] [Help] [answered]
I trying to get UGS running on a Pi4 without success. UGS is apparently launching (the normal startup screen / sequence is shown) but then opens into a blank window that only shows the UGS version at the top (2.1.6).

When opening from terminal the following errors are shown:

_OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.netbeans.TopSecurityManager (file:/home/username/Downloads/ugsplatform-linux-aarch64/platform/lib/boot.jar)
WARNING: Please consider reporting this to the maintainers of org.netbeans.TopSecurityManager
WARNING: System::setSecurityManager will be removed in a future release
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: XDG_RUNTIME_DIR is invalid or not set in the environment._

From how I understand this this points to issues with the Java version bundled with UGS.

Any idea how to fix this?

Thanks, Hubert

**Comment by breiler** [comment]
What distribution did you install on the Pi4?

Can you post your log file:
`~/.ugsplatform/var/log/messages.log`

**Reply by HubertJH** [reply]
Thanks for looking into this - very much appreciated.

The Raspberry runs:
Raspberry Pi OS with desktop and recommended software
Release date: March 15th 2024
System: 64-bit
Kernel version: 6.6
Debian version: 12 (bookworm)
Size: 2,803MB
On first boot I let the system download the latest updates. Other than that there is no other software installed.

Then I installed linux-aarch64-ugs-platform-app-2.1.6.tar.gz and extracted into the Downloads folder.
I also tried the "All Platforms" package with no success. 

With regard to the log file there is no directory .ugsplatform on my system so I cannot post a log. There are a few hidden files in various subfolders but no hidden directory of that name. Could you pls help to locate this. Or is this part of the issue?

Btw, I am running the system in super user mode.

**Reply by breiler** [reply]
You should not run as super user.

You might not have the group permissions connect to the controller with the regular user in which case you need to add your user to the dialout group, but that is a later problem. Read this article:
https://support.arduino.cc/hc/en-us/articles/360016495679-Fix-port-access-on-Linux

Start the software as a regular user, if it fails look for the log file in `~/.ugsplatform/var/log/messages.log` again.

**Comment by HubertJH** [comment]
Thanks for the advice - UGS does start normally when logged in as a regular user.

But now I am having the issue reported at https://github.com/winder/Universal-G-Code-Sender/discussions/2316. The Arduino is listed (lsusb) as BUS 001 Device 003: ID... but I only have ttyUSB0 as the only port option in ugs.
The Arduino connects fine to my Macbook.

**Reply by breiler** [reply, marked as answer]
And you've made sure that your user has the proper user rights to read the serial device with the link I gave you before:
https://support.arduino.cc/hc/en-us/articles/360016495679-Fix-port-access-on-Linux

Usually this is needed and then a reboot:
```
sudo usermod -a -G dialout <username>
```

**Reply by HubertJH** [reply]
Sorry, my bad  - did not read the article thoroughly enough. The Arduino connects fine now so this issue is resolved.

Many thanks again. And also for your incredible enthusiasm developing and supporting UGS - great work!
